scout/server/blueprints/variant/utils.py

```diff
@@ -0,0 +1,453 @@
+        if not refseq_id:
+            continue
+        transcript['refseq_id'] = refseq_id
+        variant_obj['has_refseq'] = True
+        
+        refseq_transcripts.append(transcript)
+        # Check if the refseq id are disease associated
+        if refseq_id in disease_associated_no_version:
+            transcript['is_disease_associated'] = True
+
+        # Since a ensemble transcript can have multiple refseq identifiers we add all of
+        # those
+        transcript['refseq_identifiers'] = hgnc_transcript.get('refseq_identifiers',[])
+
+        transcript['change_str'] = transcript_str(transcript, hgnc_symbol)
+    
+    variant_gene['primary_transcripts'] = refseq_transcripts
+        
+
+def add_gene_info(store, variant_obj, gene_panels=None, genome_build=None):
+    """Add extra information about genes from gene panels.
+    
+    There are are sometimes additional information that are manually curated in the gene panels.
+    This information needs to be added to the variant before sending it to the template.
+
+    This function will loop over all genes and add that extra information.
+
+    Args:
+        store(scout.adapter.MongoAdapter)
+        variant_obj(dict): A variant from the database
+        gene_panels(list(dict)): List of panels from database
+        genome_build(str)
+    
+    Returns:
+        variant_obj
+    """
+    gene_panels = gene_panels or []
+    genome_build = genome_build or '37'
+
+    # Add a variable that checks if there are any refseq transcripts
+
+    # extra_info will hold information from gene panels
+    extra_info = {}
+    for panel_obj in gene_panels:
+        for gene_info in panel_obj['genes']:
+            hgnc_id = gene_info['hgnc_id']
+            if hgnc_id not in extra_info:
+                extra_info[hgnc_id] = []
+
+            extra_info[hgnc_id].append(gene_info)
+
+    # Loop over the genes in the variant object to add information
+    # from hgnc_genes and panel genes to the variant object
+    variant_obj['has_refseq'] = False
+    variant_obj['disease_associated_transcripts'] = []
+    all_models = set()
+    for variant_gene in variant_obj.get('genes', []):
+        hgnc_id = variant_gene['hgnc_id']
+        # Get the hgnc_gene
+        hgnc_gene = store.hgnc_gene(hgnc_id, build=genome_build)
+
+        if not hgnc_gene:
+            continue
+        hgnc_symbol = hgnc_gene['hgnc_symbol']
+        # Add omim information if gene is annotated to have incomplete penetrance
+        if hgnc_gene.get('incomplete_penetrance'):
+            variant_gene['omim_penetrance'] = True
+
+        ############# PANEL SPECIFIC INFORMATION #############
+        # Panels can have extra information about genes and transcripts
+        panel_info = add_panel_specific_gene_info(extra_info.get(hgnc_id, []), variant_gene)
+        variant_gene.update(panel_info)
+        
+        update_transcripts_information(variant_gene, hgnc_gene, variant_obj)
```

Comment:
This is also not so clear: why a transcript update function here, then a section of code with transcripts being updated afterwards? Perhaps merge or make one more update function? Or just document it.

Reply:
Good idea

Reply:
AAAHHHhh it doesn't work! This informatin is added on the variant_obj and is collected from the panels. There is no nice way of solving this I'm afraid...